praw/models/reddit/subreddit.py

```diff
@@ -2150,3 +2167,45 @@ def revisions(self, **generator_kwargs):
         url = API_PATH['wiki_revisions'].format(subreddit=self.subreddit)
         return WikiPage._revision_generator(
             self.subreddit, url, generator_kwargs)
+
+
+class SubredditEmoji(object):
+    """Provides a set of moderation functions to a Subreddit for emoji."""
+
+    def __getitem__(self, name):
+        """Lazily return the Emoji for the subreddit named ``name``.
+
+        This method is to be used to fetch a specific emoji url, like so:
+
+        .. code:: python
+
+           emoji = reddit.subreddit('iama').emoji['cake']
+           print(emoji)
+
+        """
+        return Emoji(self.subreddit._reddit, self.subreddit, name.lower())
+
+    def __init__(self, subreddit):
+        """Create a SubredditModeration instance.
+
+        :param subreddit: The subreddit to moderate.
+
+        """
+        self.subreddit = subreddit
+
+    def __iter__(self):
+        """Iterate through the Emoji for the subreddit.
+
+        This method is to be used to discover all emoji for a subreddit:
+
+        .. code:: python
+
+           for emoji in reddit.subreddit('iama').emoji:
+               print(emoji)
+
+        """
+        response = self.subreddit._reddit.get(
+            API_PATH['emoji_list'].format(subreddit=self.subreddit),
+            params={'unique': self.subreddit._reddit._next_unique})
+        for emoji_name in response[self.subreddit]:
```

Comment:
I believe `self.subreddit` should be `self.subreddit.fullname`. When I run your code, I get the following exception and traceback:

```
Traceback (most recent call last):
  File "emojitest.py", line 8, in <module>
    for e in emoji:
  File "praw/models/reddit/subreddit.py", line 2210, in __iter__
    for emoji_name in response[self.subreddit]:
KeyError: Subreddit(display_name='mysub')
```

When I change it to `self.subreddit.fullname` everything works as intended.